src/containers/Tenant/Diagnostics/TopShards/Filters/Filters.tsx

```diff
@@ -0,0 +1,56 @@
+import {RadioButton} from '@gravity-ui/uikit';
+
+import {DateRange, DateRangeValues} from '../../../../../components/DateRange';
+
+import {
+    EShardsWorkloadMode,
+    IShardsWorkloadFilters,
+} from '../../../../../types/store/shardsWorkload';
+
+import {isEnum} from '../../../../../utils/typecheckers';
+
+import i18n from '../i18n';
+import {b} from '../TopShards';
+
+import './Filters.scss';
+
+interface FiltersProps {
+    value: IShardsWorkloadFilters;
+    onChange: (value: Partial<IShardsWorkloadFilters>) => void;
+    className?: string;
+}
+
+export const Filters = ({value, onChange, className}: FiltersProps) => {
+    const handleModeChange = (mode: string) => {
```

Comment:
What is the reasons of a such check, if you define both filters and handler in the same component?

Reply:
`RadioButton` doesn't allow specifying its value type, it's just `string`
`onChange` expects a restricted filters shape with `mode` being of type `EShardsWorkloadMode`
`string` can't be assigned to `EShardsWorkloadMode`
this check narrows incoming value type